src/components/UserSelector.tsx

```diff
@@ -1,6 +1,29 @@
-import React from 'react';
+import { Dispatch, FC, SetStateAction, useState } from 'react';
+import cn from 'classnames';
+
+import { User } from '../types/User';
+
+interface Props {
+  users: User[];
+  userSelected: User | null;
+  setUserSelected: Dispatch<SetStateAction<User | null>>;
+  getUsersPost: (userId: User['id']) => void;
+}
+
+export const UserSelector: FC<Props> = ({
+  users,
+  userSelected,
+  setUserSelected,
+  getUsersPost,
+}) => {
+  const [isClicked, setIsClicked] = useState(false);
+
+  const handleSelectUser = (user: User) => {
+    setUserSelected(user);
+    setIsClicked(false);
+    getUsersPost(user.id);
+  };
 
-export const UserSelector: React.FC = () => {
   return (
     <div data-cy="UserSelector" className="dropdown is-active">
```

Comment:
The class `is-active` is always applied to the dropdown, which might not be the intended behavior. Consider conditionally applying this class based on the `isClicked` state to properly toggle the dropdown visibility.